app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailViewImpl.kt

```diff
@@ -47,6 +52,24 @@ class NoteDetailViewImpl(val bundle: Bundle) : BaseViewImpl<NoteDetailView, Note
 
     override fun getLayoutId(): Int = R.layout.note_detail_view
 
+    override fun onSaveViewState(view: View, outState: Bundle) {
+        super.onSaveViewState(view, outState)
+        outState.putString(STATE_NOTE_TITLE, noteTitleTextView.text.toString())
+        outState.putString(STATE_NOTE_BODY, noteBodyTextView.text.toString())
+        outState.putString(STATE_NOTE_CREATED, noteDateCreated.text.toString())
+        outState.putString(STATE_NOTE_UPDATED, noteDateUpdated.text.toString())
```

Comment:
Is this possibly because you try to reinitialize it with data? You should probably check the savedInstanceState's existence before deciding whether to load the model data.